safekit-cluster-mirror/nestedtemplates/cluster.json

```diff
@@ -0,0 +1,650 @@
+				"description": "loadbalancer (optional. If set, a loadbalancer will be created, with the VIP as frontend and the VMs in the backend pool."
+			},
+			"defaultValue": "none",
+			"allowedValues": [
+				"External",
+				"none"
+			]
+		},
+		"azurePwsh": {
+			"type": "string",
+			"metadata": {
+				"description": "install azure powershell module (optional)"
+			},
+			"defaultValue": "no",
+			"allowedValues": [
+				"yes",
+				"no"
+			]
+		},
+		"location": {
+			"type": "string",
+			"metadata": {
+				"description": "resources location"
+			},
+			"defaultValue": "[resourceGroup().location]"
+		},
+		"safekitFileUri": {
+			"type": "string",
+			"metadata": {
+				"description": "url of safekit package"
+			},
+			"defaultValue": ""
+		},
+		"_artifactsLocation": {
+			"type": "string",
+			"metadata": {
+				"description": "base URL of deployment resources (template,subtemplates,scripts)"
+			},
+			"defaultValue": "[deployment().properties.templateLink.uri]"
+		},
+		"_artifactsLocationSasToken": {
+			"type": "securestring",
+			"metadata": {
+				"description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated. Use the defaultValue if the staging location is not secured."
+			},
+			"defaultValue": ""
+		}
+	},
+	"variables": {
+		"allvms": [
+			"VM1",
+			"VM2",
+			"VM3",
+			"VM4"
+		],
+		"vms": "[take(variables('allvms'),parameters('clusterNodes'))]",
+		"cltemplate": "[uri(parameters('_artifactsLocation'),'nested/cluster.json')]",
+		"frontEndIPConfigName": "[concat(parameters('VIPDnsLabel'),'LBFE')]",
+		"lbPoolName": "[concat(parameters('VIPDnsLabel'),'BEP')]",
+		"lbProbeName": "[concat(parameters('VIPDnsLabel'),'LBP')]",
+		"lbName": "[concat(parameters('VIPDnsLabel'),'LB')]",
+		"NetworkContributor": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/','4d97b98b-1d4f-4787-a291-c67834d212e7')]",
+		"ostype": "[if(equals(parameters('OS'), 'Linux CentOS'),'linux','windows')]",
+		"adminUsername": "[if(empty(parameters('adminUser')),concat('admin',resourceGroup().name),parameters('adminUser'))]",
+		"ips": [
+			"10.0.0.10",
+			"10.0.0.11",
+			"10.0.0.12",
+			"10.0.0.13"
+		],
+		"sktemplate": "[concat(uri(parameters('_artifactsLocation'), 'nestedtemplates/safekitdepl.json'),parameters('_artifactsLocationSasToken'))]",
+		"skclustertemplate": "[concat(uri(parameters('_artifactsLocation'), 'nestedtemplates/installCluster.json'),parameters('_artifactsLocationSasToken'))]",
+		"createviptemplate": "[concat(uri(parameters('_artifactsLocation'),'nestedtemplates/createvip.json'),parameters('_artifactsLocationSasToken'))]",
+		"skmoduletemplate": "[concat(uri(parameters('_artifactsLocation'),'nestedtemplates/installModule.json'),parameters('_artifactsLocationSasToken'))]",
+		"NSGName": "[concat(uniquestring(resourceGroup().id,parameters('location')),'-NSG')]",
```

Comment:
this does not need to be globally unique, just use a literal value here

Reply:
OK